assets/list_assets.py

```diff
@@ -0,0 +1,221 @@
+import sublime_plugin
+import sublime
+import mdpopups
+import re
+from typing import Dict
+
+from pieces_os_client import *
+from pieces.settings import PiecesSettings
+
+
+
+
+class PiecesListAssetsCommand(sublime_plugin.WindowCommand):
+	sheets_md = {}
+	def run(self,pieces_asset_id):
+
+		api_instance = AssetApi(PiecesSettings.api_client)
+		api_response = api_instance.asset_specific_asset_export(pieces_asset_id, "MD")
+		
+		markdown_text = api_response.raw.string.raw
+
+		markdown_text_table = tabulate_from_markdown(markdown_text)
+
+		sheet = mdpopups.new_html_sheet(self.window,api_response.name,markdown_text_table,css = ".div_wrapper {margin-left:2rem}",wrapper_class="div_wrapper")
+		
+		
+		sheet_id = sheet.id()
+		code_block_pattern = r'(```[\s\S]*?\n```)'
+
+
+		# Find all code blocks
+		code_block = re.findall(code_block_pattern, markdown_text)
+		try:
+			iterable = AssetSnapshot.assets_snapshot[pieces_asset_id].formats.iterable
+			sorted_iterable = sorted(iterable, key=lambda x: x.updated.value, reverse=True)
+			language = sorted_iterable[0].analysis.code.language
+		except:
+			language = None
+		PiecesListAssetsCommand.sheets_md[sheet_id] = {"code":"\n".join(code_block[0].split("\n")[1:-1]),"name":api_response.name,"language":language,"id":pieces_asset_id}
+
+
+	def input(self,args):
+		return PiecesAssetIdInputHandler()
+
+
+
+
+
+
+class PiecesHandleMarkdownCommand(sublime_plugin.WindowCommand):
```

Comment:
each one of these classes should be in their own files

Reply:
but this is related to the list_asset comment it is the input search for the assets in the open assets